src/cmd/manage_nsfs.js

```diff
@@ -0,0 +1,411 @@
+    manage_bucket_operations(action, data, config_root);
+}
+
+async function fetch_bucket_data(argv) {
+    const action = argv._[1] || '';
+    const config_root = argv.config_root ? String(argv.config_root) : '';
+    const name = argv.name && new SensitiveString(String(argv.name));
+    const email = argv.email && new SensitiveString(String(argv.email));
+    const from_file = argv.from_file ? String(argv.from_file) : '';
+    let data;
+    if (from_file) {
+        const raw_data = await fs.promises.readFile(String(argv.from_file));
+        data = JSON.parse(raw_data.toString());
+        data.name = new SensitiveString(data.name);
+    }
+    if (!data) {
+        const bucket_path = argv.path ? String(argv.path) : '';
+        data = {
+            name: name,
+            system_owner: email,
+            bucket_owner: email,
+            tag: "",
+            versioning: "DISABLED",
+            path: bucket_path,
+            should_create_underlying_storage: true,
+        };
+    }
+    if (action === 'update') {
+        data = _.omitBy(data, _.isUndefined);
+        data = update_bucket(config_root, data);
+    }
+    return data;
+}
+
+async function update_bucket(config_root, target) {
+    const account_path = config_root + '/buckets';
+    let source;
+    try {
+        const source_raw = await fs.promises.readFile(path.join(account_path, target.name + '.json'));
+        source = JSON.parse(source_raw.toString());
+        //target.nsfs_account_config.new_buckets_path = target.nsfs_account_config.new_buckets_path || source.nsfs_account_config.new_buckets_path;
+    } catch (err) {
+        console.error('NSFS Manage command: Could not find bucket ' + target.name + ' to update');
+        print_account_usage();
+    }
+    const data = _.merge({}, source, target);
+    return data;
+}
+
+async function manage_bucket_operations(action, data, config_root) {
+    const bucket_config_path = config_root + '/buckets';
+    if (action === 'add') {
+        const is_valid = await validate_bucket_add_args(data);
+        if (!is_valid) {
+            print_bucket_usage();
+            return;
+        }
+        create_resourse(data.name, bucket_config_path, data);
+    } else if (action === 'status') {
+        const is_valid = await validate_minimum_bucket_args(data);
+        if (!is_valid) {
+            print_account_usage();
+            return;
+        }
+        const bucket_dir_stat = await fs_utils.file_exists(path.join(bucket_config_path, data.name + '.json'));
+        if (bucket_dir_stat) {
+            console.log("Bucket exists with access key : " + data.name);
+        } else {
+            console.log("Bucket do not exists with access key : " + data.name);
+        }
+    } else if (action === 'update') {
+        const is_valid = await validate_bucket_add_args(data);
+        if (!is_valid) {
+            print_bucket_usage();
+        }
+        create_resourse(data.name, bucket_config_path, data);
```

Comment:
why do we create the resource on update? also I saw the on fetch we update the file

Reply:
@romayalon its calling method `await fs_utils.replace_file(full_path, data);`. It creates the file if the file missing and replaces already exists. On fetch we we are just getting data from the config file and updating object with input data.